### PR TITLE
Adds "Use Exact Amounts" option to Reactions

### PR DESCRIPTION
**Pull request by VernaSaurus**
allows you to make reactions require exact amounts rather than ratios, as shown with the dough and cake batter recipes

also i'm back haha gottem

### CODE REVIEW

File: UnityProject/Assets/Scripts/Chemistry/Reaction.cs [hunk 1]
```diff
@@ -33,28 +34,62 @@ public bool Apply(MonoBehaviour sender, ReagentMix reagentMix)
...
+					var reactionResult = reactionAmount * result.Value;
+					reagentMix.Add(result.Key, reactionResult);
+				}
+
+				foreach (var effect in effects)
+				{
+					effect.Apply(sender, reactionAmount);
+				}
+			}
+			else if (useExactAmounts == true)
```
Comment (vlandemart) [inline UnityProject/Assets/Scripts/Chemistry/Reaction.cs]: 90% of this else is just copypaste of if statement
Can't you just insert
`if (useExactAmounts && !ingredients.All(ingredient => reagentMix[ingredient.Key] >= ingredient.Value))
    return;`
after if statement at line 41 not to make unneeded nesting?

**Review by vlandemart — CHANGES_REQUESTED** [review]

**Comment by VernaSaurus** [comment]
I noticed that only after I already closed everything out, I'll fix it

**Comment by VernaSaurus** [comment]
alright, i did it

**Comment by ThatDan123** [comment]
image

I would suggest making the bool (useExactAmounts) the first condition of the If statement just for performance reasons,  unless a code reviewer doesnt think that'll be necessary.

File: UnityProject/Assets/Scripts/Chemistry/Reaction.cs [hunk 2]
```diff
@@ -33,6 +34,12 @@ public bool Apply(MonoBehaviour sender, ReagentMix reagentMix)
 			{
 				return false;
 			}
+
+
+			if (!ingredients.All(ingredient => reagentMix[ingredient.Key] >= ingredient.Value) && useExactAmounts == true)
+			{
+				return false;
+			}
```
Comment (RedlineTriad) [inline UnityProject/Assets/Scripts/Chemistry/Reaction.cs]: This will use inexact amounts for when there is more of an ingredient than is strictly needed.

**Comment by Bod9001** [comment]
all good?

**Comment by VernaSaurus** [comment]
yep

**Pull request merged**